lib/kube/kubeconfig/kubeconfig.go

```diff
@@ -23,38 +26,82 @@ var log = logrus.WithFields(logrus.Fields{
 
 // Values are Teleport user data needed to generate kubeconfig entries.
 type Values struct {
-	// Name is used to name kubeconfig sections ("context", "cluster" and
+	// TeleportClusterName is used to name kubeconfig sections ("context", "cluster" and
 	// "user"). Should match Teleport cluster name.
-	Name string
+	TeleportClusterName string
 	// ClusterAddr is the public address the Kubernetes client will talk to,
 	// usually a proxy.
 	ClusterAddr string
 	// Credentials are user credentials to use for authentication the
 	// ClusterAddr. Only TLS fields (key/cert/CA) from Credentials are used.
 	Credentials *client.Key
+	// TshBinaryPath is a path to the tsh binary for use as exec plugin.
+	//
+	// If not set, static key/cert from Credentials are written to kubeconfig
+	// instead.
+	TshBinaryPath string
```

Comment:
If these attributes are only applicable with TshBinaryPath set - have you considered grouping them into a sub-struct?
```go
type Values struct {
	// ...
	Exec *TshExecPlugin
}
```